idunn/api/instant_answer.py

```diff
@@ -71,10 +71,36 @@ class InstantAnswerResponse(BaseModel):
     data: InstantAnswerData
 
 
-NoInstantAnswerToDisplay = HTTPException(status_code=404, detail="No instant answer to display")
+class NoInstantAnswerToDisplay(HTTPException):
+    def __init__(self, query=None, lang=None):
+        super().__init__(status_code=404, detail="No instant answer to display")
+        if query is not None:
+            logger.info(
+                "get_instant_answer: no answer",
+                extra={
+                    "request": {
+                        "query": query,
+                        "lang": lang,
+                    }
+                },
+            )
```

Comment:
Isn't it a bit odd to log from the exception's constructor? `query` and `lang` are not actually part of the exception data, that feels a bit confusing.

I'm not sure, but maybe raising the exception from an explicit function or method would make more sense ?

Reply:
You're right. I was a bit reluctant to change how exception was raised, but logging from the constructor is probably not a good pattern.

The exception is now raised from a function `no_instant_answer` is defined. This led me to to some refactoring: "get_instant_answer" was a very long function, and the linter complained about too many `return`. 

So it should be more readable now. What do you think ?